compiler/ballerina-lang/src/main/java/io/ballerina/projects/directory/BuildProject.java

```diff
@@ -207,18 +208,30 @@ public void save() {
             writeBuildFile(buildFilePath);
             writeDependencies();
         } else {
-            // build file exists
-            BuildJson buildJson = readBuildJson(buildFilePath);
-            // need to update Dependencies toml
-            writeDependencies();
-            // check whether last updated time is expired
-            if (shouldUpdate) {
-                // update build json file
-                writeBuildFile(buildFilePath);
-            } else {
-                // only update build time
-                buildJson.setLastBuildTime(System.currentTimeMillis());
-                ProjectUtils.writeBuildFile(buildFilePath, buildJson);
+            try {
+                // build file exists
+                BuildJson buildJson = readBuildJson(buildFilePath);
+                // need to update Dependencies toml
+                writeDependencies();
+                if (buildJson != null) {
+                    // check whether last updated time is expired
+                    if (shouldUpdate) {
+                        // update build json file
+                        writeBuildFile(buildFilePath);
+                    } else {
+                        // only update build time
+                        buildJson.setLastBuildTime(System.currentTimeMillis());
+                        ProjectUtils.writeBuildFile(buildFilePath, buildJson);
+                    }
+                } else {
+                    Files.deleteIfExists(buildFilePath);
+                    createBuildFile(buildFilePath);
```

Comment:
@azinneera This particular line fails on windows with an IO Exception as specified in the build scan. 

```
at io.ballerina.projects.directory.BuildProject.createBuildFile(BuildProject.java:404)
at io.ballerina.projects.directory.BuildProject.save(BuildProject.java:228)
at io.ballerina.projects.test.PackageResolutionTests.testProjectSaveWithEmptyBuildFile(PackageResolutionTests.java:166)
```

So if an empty build file exists, we delete it, create a new one and rewrite it. Is this an acceptable way to proceed with such a case?